### PR TITLE
An 4825/raydium swaps

### PR DESCRIPTION
**Pull request by tarikceric**
New swaps models for Raydium using decoded instructions

- FR on dev-2xl
```
22:01:40  3 of 3 OK created sql incremental model silver.swaps_intermediate_raydium_stable  [SUCCESS 1 in 269.03s]
22:06:07  2 of 3 OK created sql incremental model silver.swaps_intermediate_raydium_clmm . [SUCCESS 1 in 535.87s]
22:23:38  1 of 3 OK created sql incremental model silver.swaps_intermediate_raydium_amm .. [SUCCESS 1 in 1586.88s]
```
- test
```
18:33:01  Finished running 61 tests, 11 hooks in 0 hours 3 minutes and 33.46 seconds (213.46s).
18:33:02  Completed successfully
18:33:02  Done. PASS=61 WARN=0 ERROR=0 SKIP=0 TOTAL=61
```

### CODE REVIEW

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/swaps/silver__swaps_intermediate_raydium_amm.sql [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        tx_id,
+        INDEX,
+        inner_index,
+        COALESCE(LEAD(inner_index) over (PARTITION BY tx_id, INDEX
+    ORDER BY
+        inner_index) -1, 999999) AS inner_index_end,
+        program_id,
+                silver.udf_get_account_pubkey_by_name('userSourceOwner', decoded_instruction:accounts) as temp_user_source_owner,
+        CASE
+            WHEN temp_user_source_owner IS NOT NULL then silver.udf_get_account_pubkey_by_name('userSourceOwner', decoded_instruction:accounts)
```
[the remark this answers is not in the document]
Comment (tarikceric) [reply models/silver/swaps/silver__swaps_intermediate_raydium_amm.sql:75]: good catch, updated

File: models/silver/swaps/silver__swaps_intermediate_raydium_amm.sql [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            else silver.udf_get_account_pubkey_by_name('uerSourceTokenAccount', decoded_instruction:accounts)
+        END AS destination_token_account,
+        case WHEN temp_user_source_owner IS NOT NULL and event_type = 'swapBaseOut' then silver.udf_get_account_pubkey_by_name('poolPcTokenAccount', decoded_instruction:accounts)
+            else silver.udf_get_account_pubkey_by_name('poolCoinTokenAccount', decoded_instruction:accounts) 
+            end as program_destination_token_account,
+        CASE
+            WHEN  temp_user_source_owner IS NOT NULL and event_type = 'swapBaseOut' then silver.udf_get_account_pubkey_by_name('poolCoinTokenAccount', decoded_instruction:accounts)
+            WHEN temp_user_source_owner IS NOT NULL and event_type = 'swapBaseIn' then silver.udf_get_account_pubkey_by_name('poolPcTokenAccount', decoded_instruction:accounts)
+            else silver.udf_get_account_pubkey_by_name('ammTargetOrders', decoded_instruction:accounts)
+        END AS program_source_token_account,
```
[the remark this answers is not in the document]
Comment (tarikceric) [reply models/silver/swaps/silver__swaps_intermediate_raydium_amm.sql:95]: updated

File: models/silver/swaps/silver__swaps_intermediate_raydium_amm.sql [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        )
+        LEFT JOIN transfers e
+        ON A.tx_id = e.tx_id
+        AND A.destination_token_account = e.dest_token_account
+        AND A.program_source_token_account = e.source_token_account
+        AND A.index = e.index_1
+        AND (
+            (e.inner_index_1 BETWEEN A.inner_index AND A.inner_index_end)
+            OR A.inner_index IS NULL
+        )
```
[the remark this answers is not in the document]
Comment (tarikceric) [reply models/silver/swaps/silver__swaps_intermediate_raydium_amm.sql:187]: added comments

File: models/silver/swaps/silver__swaps_intermediate_raydium_amm.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        tests: 
+          - not_null: *recent_date_filter
+      - name: TO_MINT
+        description:  "{{ doc('swaps_to_mint') }}"
+        tests: 
+          - not_null: *recent_date_filter
+      - name: SWAP_INDEX
+        description: "{{ doc('swaps_swap_index') }}"
+        tests: 
+          - not_null: *recent_date_filter
```
[the remark this answers is not in the document]
Comment (tarikceric) [reply models/silver/swaps/silver__swaps_intermediate_raydium_amm.yml:77]: added to `amm` and `stable`

**Review by desmond-hui — APPROVED** [review]

**Pull request merged**